src/Orleans.Transactions/State/TransactionQueue.cs

```diff
@@ -4,338 +4,215 @@
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.Extensions.Logging;
-using Orleans;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
 using Orleans.Runtime;
 using Orleans.Transactions.Abstractions;
-using Orleans.Transactions;
 using Orleans.Storage;
+using Orleans.Configuration;
 
-namespace Orleans.Transactions
+namespace Orleans.Transactions.State
 {
-    /// <summary>
-    /// Stateful facet that respects Orleans transaction semantics
-    /// </summary>
-    /// <typeparam name="TState"></typeparam>
-    public partial class TransactionalState<TState> : ITransactionalState<TState>, ITransactionParticipant, ILifecycleParticipant<IGrainLifecycle>
+    internal class TransactionQueue<TState>
         where TState : class, new()
     {
-        private bool detectReentrancy;
-
-
-        private async Task StorageWork()
+        private readonly TransactionalStateOptions options;
+        private readonly ITransactionParticipant resource;
+        private readonly Action deactivate;
+        private readonly ITransactionalStateStorage<TState> storage;
+        private readonly JsonSerializerSettings serializerSettings;
+        private readonly BatchWorker storageWorker;
+        private readonly BatchWorker confirmationWorker;
+        private readonly ILogger logger;
+        private readonly Dictionary<Guid, TransactionRecord<TState>> confirmationTasks;
+        private CommitQueue<TState> commitQueue;
```

Comment:
This line irks me so much. The field is never assigned. This only works because `CommitQueue<TState>` is a struct. It's unintuitive that this could be correct.